cpp/src/strings/datetime.cu

```diff
@@ -0,0 +1,691 @@
+        if( units==timestamp_units::hours )
+            return (days*24L) + hour + (tzadjust/60);
+
+        auto minute = timeparts[TP_MINUTE];
+        if( units==timestamp_units::minutes )
+            return static_cast<int64_t>(days * 24L * 60L) + (hour * 60L) + minute + tzadjust;
+
+        auto second = timeparts[TP_SECOND];
+        int64_t timestamp = (days * 24L * 3600L) + (hour * 3600L) + (minute * 60L) + second + (tzadjust*60);
+        if( units==timestamp_units::seconds )
+            return timestamp;
+
+        auto subsecond = timeparts[TP_SUBSECOND];
+        if( units==timestamp_units::ms )
+            timestamp *= 1000L;
+        else if( units==timestamp_units::us )
+            timestamp *= 1000000L;
+        else if( units==timestamp_units::ns )
+            timestamp *= 1000000000L;
+        timestamp += subsecond;
+        return timestamp;
+    }
+
+     __device__ int64_t operator()(size_type idx)
+    {
+        if( d_strings.is_null(idx) )
+            return 0;
+        string_view d_str = d_strings.element<string_view>(idx);
+        if( d_str.empty() )
+            return 0;
+        //
+        int32_t timeparts[TP_ARRAYSIZE] = {0,1,1}; // month and day are 1-based
+        if( parse_into_parts(d_str,timeparts) )
+            return 0; // unexpected parse case
+        //
+        return timestamp_from_parts(timeparts,units);
+    }
+};
+
+} // namespace
+
+//
+std::unique_ptr<cudf::column> to_timestamps( strings_column_view strings,
+                                             timestamp_units units,
+                                             const char* format,
+                                             rmm::mr::device_memory_resource* mr,
+                                             cudaStream_t stream )
+{
+    size_type strings_count = strings.size();
+    if( strings_count==0 )
+        return std::make_unique<column>( data_type{INT64}, 0,
+                          rmm::device_buffer{0,stream,mr}, // data
+                          rmm::device_buffer{0,stream,mr}, 0 ); // nulls}
+
+    CUDF_EXPECTS( format!=nullptr, "Format parameter must not be nullptr");
+
+    format_compiler compiler(format,units);
+    format_program* d_prog = compiler.compile_to_device();
+
+    auto execpol = rmm::exec_policy(stream);
+    auto strings_column = column_device_view::create(strings.parent(), stream);
+    auto d_column = *strings_column;
+
+    // copy null mask
+    rmm::device_buffer null_mask;
+    cudf::size_type null_count = d_column.null_count();
+    if( d_column.nullable() )
+        null_mask = rmm::device_buffer( d_column.null_mask(),
+                                        gdf_valid_allocation_size(strings_count),
+                                        stream, mr);
+    // create output column
+    auto results = std::make_unique<cudf::column>( cudf::data_type{cudf::INT64}, strings_count,
+        rmm::device_buffer(strings_count * sizeof(int32_t), stream, mr),
```

Comment:
Isn't `strings_count * sizeof(int32_t)` going to be half the size if the column type is `int64`, or am I missing something about `strings_count`?
```suggestion
        rmm::device_buffer(strings_count * sizeof(int64_t), stream, mr),
```